Add value argument to URLSearchParams's has() and delete()

annevk opened the pull request:
#### 93a5678f2d4f370aa8f957928f96fbb501e9e168
<pre>
Add value argument to URLSearchParams&apos;s has() and delete()
<a href="https://bugs.webkit.org/show_bug.cgi?id=256369">https://bugs.webkit.org/show_bug.cgi?id=256369</a>
rdar://108949109

Reviewed by Chris Dumez.

Aligns WebKit with a recent URLSearchParams change.

Also syncs WPT url/ and new tests are upstreamed via <a href="https://github.com/web-platform-tests/wpt/pull/39865.">https://github.com/web-platform-tests/wpt/pull/39865.</a>

* LayoutTests/imported/w3c/web-platform-tests/url/a-element-origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element-origin-xhtml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element-xhtml_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/resources/urltestdata.json:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any.worker_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any.worker-expected.txt:
* Source/WebCore/html/URLSearchParams.cpp:
(WebCore::URLSearchParams::has const):
(WebCore::URLSearchParams::remove):
* Source/WebCore/html/URLSearchParams.h:
(WebCore::URLSearchParams::remove):
(WebCore::URLSearchParams::has const):
* Source/WebCore/html/URLSearchParams.idl:

Canonical link: <a href="https://commits.webkit.org/263726@main">https://commits.webkit.org/263726@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4eef375a8c9f5005d29be0db81c10641e823cfb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5582 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5735 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5923 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7134 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5581 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/5582 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5960 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5708 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/7271 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5685 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5746 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5001 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7172 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3224 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5020 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/12145 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5090 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5099 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/6908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5537 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4532 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4990 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1322 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9095 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5351 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/URLSearchParams.h:
```diff
@@ -47,10 +47,12 @@ class URLSearchParams : public RefCounted<URLSearchParams> {
 
     void append(const String& name, const String& value);
-    void remove(const String& name);
+    void remove(const String& name, const String& value);
```
Can use `= { }` for the second parameter.

cdumez commented on Source/WebCore/html/URLSearchParams.h:
```diff
@@ -47,10 +47,12 @@ class URLSearchParams : public RefCounted<URLSearchParams> {
     String get(const String& name) const;
     Vector<String> getAll(const String& name) const;
-    bool has(const String& name) const;
+    bool has(const String& name, const String& value) const;
```
Can use `= { }` for the second parameter.

cdumez commented on Source/WebCore/html/URLSearchParams.h:
```diff
@@ -47,10 +47,12 @@ class URLSearchParams : public RefCounted<URLSearchParams> {
     void append(const String& name, const String& value);
-    void remove(const String& name);
+    void remove(const String& name, const String& value);
+    void remove(const String& name) { remove(name, nullString()); }
```
Not needed.

cdumez commented on Source/WebCore/html/URLSearchParams.h:
```diff
@@ -47,10 +47,12 @@ class URLSearchParams : public RefCounted<URLSearchParams> {
     Vector<String> getAll(const String& name) const;
-    bool has(const String& name) const;
+    bool has(const String& name, const String& value) const;
+    bool has(const String& name) const { return has(name, nullString()); }
```
Not needed.

cdumez reviewed (approved):
LGTM

annevk committed:
Add value argument to URLSearchParams's has() and delete()
https://bugs.webkit.org/show_bug.cgi?id=256369
rdar://108949109

Reviewed by Chris Dumez.

Aligns WebKit with a recent URLSearchParams change.

Also syncs WPT url/ and new tests are upstreamed via https://github.com/web-platform-tests/wpt/pull/39865.

* LayoutTests/imported/w3c/web-platform-tests/url/a-element-origin-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element-origin-xhtml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element-xhtml_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/resources/urltestdata.json:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any.worker_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-delete.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/url/urlsearchparams-has.any.worker-expected.txt:
* Source/WebCore/html/URLSearchParams.cpp:
(WebCore::URLSearchParams::has const):
(WebCore::URLSearchParams::remove):
* Source/WebCore/html/URLSearchParams.h:
(WebCore::URLSearchParams::remove):
(WebCore::URLSearchParams::has const):
* Source/WebCore/html/URLSearchParams.idl:

Canonical link: https://commits.webkit.org/263726@main
